backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/GasStationService.java

```diff
@@ -102,6 +102,23 @@ private Availability getAvailabilityByType(Set<Availability> availabilities, Int
         return null;
     }
 
+    public void updateAvailability(Integer gasStationId, Integer idFuelType, Integer idStatus) {
+        GasStation gasStation = gasStationRepository.findById(gasStationId)
+                .orElseThrow(() -> new RuntimeException("GasStation not found"));
+
+        if (idFuelType == null || idFuelType <= 0 || idStatus == null || idStatus <= 0) {
+            throw new RuntimeException("FuelType or Status not valid");
+        }
+
+        Availability availability = new Availability();
```

Comment:
Change this logic to check if the availability for this fuel type and this gas station exists, if so update the status and the date, if not create a new record